flash_update/test_views.py

```diff
@@ -340,3 +341,25 @@ def test_flash_update_share(self, send_notification, render_to_pdf):
         self.assertTrue(render_to_pdf.assert_called)
         # check if send_notifications function is called.
         self.assertTrue(send_notification.assert_called)
+
+    @mock.patch('flash_update.utils.render_to_pdf')
+    def test_flash_update_pdf_export(self, render_to_pdf):
+        render_to_pdf.return_value = {
+            'filename': "test.pdf",
+            'file': b'pdf content'
+        }
+        self.client.force_authenticate(user=self.user)
+        with self.capture_on_commit_callbacks(execute=True):
+            response = self.client.post('/api/v2/flash-update/', self.body, format='json').json()
+        flash_update = FlashUpdate.objects.get(id=response['id'])
+        with self.capture_on_commit_callbacks(execute=True):
+            response = self.client.get(f'/api/v2/export-flash-update/{flash_update.id}/', format='json')
+        content = response.json()
+        self.assertEqual(content['status'], "pending")
+        self.assertEqual(content['url'], None)
+        time.sleep(3)
```

Comment:
why?

Reply:
when we try to export flahupdate it requires some time to generate pdf, so when pdf  is not ready
it will response { status: 'pending', url: null }, it takes 2,3 sec to generate pdf, then after the pdf generation it will response { status: 'ready', url: '<url_for_the_pdf>' }

Reply:
Since we have CELERY_TASK_ALWAYS_EAGER enabled on the test cases. The celery task is run after the capture_on_commit_callbacks scope. So you don't need to wait for it.

Reply:
For eg:

        with self.capture_on_commit_callbacks(execute=True):
            response = self.client.get(f'/api/v2/export-flash-update/{flash_update.id}/', format='json')
            # Here lambda: export_tasks is under transaction commit queue
       # Now the transaction commit is triggered -> with  CELERY_TASK_ALWAYS_EAGER, celery task is also triggered